### PR TITLE
Add Core topics section to all topics page

### PR DESCRIPTION
**Pull request by Will-Howard**
Screenshot 2023-02-07 at 15 14 20
Screenshot 2023-02-07 at 15 14 33
Screenshot 2023-02-07 at 15 15 10
Screenshot 2023-02-07 at 15 16 07
Screenshot 2023-02-07 at 15 16 40

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201302964208280/1203921060620451) by [Unito](https://www.unito.io)

### CODE REVIEW

**Comment by jpaddison3** [comment]
Ah jeeze. I really think this is off-center and too narrow and both of those could be fixed by removing the table of contents. Lizka will hate me but I think we should do it. (cc @agnesstenlund)

**Comment by jpaddison3** [comment]
Reminder to link to [designs](https://cea-core.slack.com/archives/C03H4KJ8J9G/p1674773143241129).

File: packages/lesswrong/components/tagging/CoreTagsSection.tsx [hunk 1]
```diff
@@ -0,0 +1,45 @@
+import React from 'react';
```
Comment (jpaddison3) [inline packages/lesswrong/components/tagging/CoreTagsSection.tsx:1]: Absolute beauty of a component. It is simple, but you did the Right Thing with the grid and it's all very :chefs-kiss:

File: packages/lesswrong/components/tagging/AllTagsPage.tsx [hunk 2]
```diff
@@ -74,7 +85,18 @@ const AllTagsPage = ({classes}: {
...
     <AnalyticsContext pageContext="allTagsPage">
       <div className={classes.root}>
-        <div className={classes.topSection}>
+        {isEAForum && <div className={classes.coreTagsSection}>
+          <ToCColumn
+            tableOfContents={<div/>}
+            header={<SectionTitle title={`Core ${taggingNamePluralSetting.get()}`} noTopMargin />}
+            centerEarly
+          >
+            <AnalyticsContext pageSectionContext="coreTagsSection">
```
Comment (jpaddison3) [inline packages/lesswrong/components/tagging/AllTagsPage.tsx:94]: [Nit] Move this analytics context inside the `CoreTagsSection`. See this [underappreciated beauty of a slack message](https://cea-core.slack.com/archives/C75BWSNBH/p1673042350642009) for why the pattern of outside-the-component `AnalyticsContext`s emerged. Once we understand the better way to do things however, we realize that it's more sensible for them to be inside the component.

File: packages/lesswrong/components/tagging/AllTagsPage.tsx [hunk 3]
```diff
@@ -45,6 +45,17 @@ const styles = (theme: ThemeType): JssStyles => ({
   addTagButton: {
     verticalAlign: "middle",
   },
+  coreTagsSection: {
+    /**
+     * I'm slightly abusing the ToCColumn component by using an empty table of contents for the top core tags section.
+     * This makes it way easier to line up with the rest of the page, while keeping the table of contents for the
+     * lower wiki section. This works fine apart from the fact that the table of contents component has a fixed height of 80vh,
+     * this removes that
+     */
```
Comment (jpaddison3) [inline packages/lesswrong/components/tagging/AllTagsPage.tsx:54]: This is not a docstring
```suggestion
    // I'm slightly abusing the ToCColumn component by using an empty table of contents for the top core tags section.
    // This makes it way easier to line up with the rest of the page, while keeping the table of contents for the
    // lower wiki section. This works fine apart from the fact that the table of contents component has a fixed height of 80vh,
    // this removes that
 ```
 Very good comment though.

File: packages/lesswrong/components/posts/TableOfContents/ToCColumn.tsx [hunk 4]
```diff
@@ -128,14 +150,15 @@ export const ToCColumn = ({tableOfContents, header, welcomeBox, children, classe
...
+        [classes.tocActivatedCenterEarly]: centerEarly,
         [classes.sideCommentsActive]: sideCommentsActive,
       }
     )}>
       <div className={classes.header}>
         {header}
       </div>
       {tableOfContents && <div className={classes.toc}>
-        <div className={classes.stickyBlockScroller}>
+        <div className={classNames(classes.stickyBlockScroller, classes.stickyBlockScrollerCenterEarly)}>
```
Comment (jpaddison3) [inline packages/lesswrong/components/posts/TableOfContents/ToCColumn.tsx:161]: I believe you mean to be conditionally applying this class and the fact that you're not is the cause of this bug:

image

**Comment by jpaddison3** [comment]
It still seems pretty nuts / bad to me that we have a full screen's worth of stuff of just core topics. you have to scroll down to notice that there's anything else on this page. @agnesstenlund help me out. I know this was your design, but maybe we can massage it somehow?

**Review by jpaddison3 — COMMENTED** [review]
Found a bug. Would also love to hear back from @agnesstenlund about the designs, and maybe we end up chopping the ToC on this page.

Maybe down the line we build a custom ToC that navigates up and down _all_ the sections, not just the ones in the ckedited ones.

**Comment by agnesstenlund** [comment]
This is really nice work @Will-Howard 👏  a lot of attention to detail.

I'm weirdly not as bothered as @jpaddison3 about the off-centeredness, but it does feel weirder than expected that the ToC only shows up below, as if the first part of the page didn't count. So I'd also be in favour of killing the ToC, or moving it up (which I assume is pretty complex).

Looks like it's now ordered by name, I think we may want to keep the order as in the excel sheet/Figma? cc @SharangP ([link here](https://www.figma.com/file/OD4vJlLuWriHqG1WJXY4LV?node-id=447:6036#361409791) about the ordering)

> It still seems pretty nuts / bad to me that we have a full screen's worth of stuff of just core topics. you have to scroll down to notice that there's anything else on this page. @agnesstenlund help me out. I know this was your design, but maybe we can massage it somehow?

I agree with this. It's more confusing than I anticipated. I do like the size of the cards as they are (I tried some slimmer versions too), so I'd suggest only showing 4 lines and then "Show 7 more" or "Show all"
Screenshot 2023-02-08 at 12 27 41

**Comment by agnesstenlund** [comment]
one small thing: would be nicer to truncate the long names in order to not have this line-break which makes the alignment and height different for some cards:
Screenshot 2023-02-08 at 12 31 18

so just go "Biosecurity and pandemic...". Not optimal, but better I think

**Comment by Will-Howard** [comment]
> one small thing: would be nicer to truncate the long names in order to not have this line-break which makes the alignment and height different for some cards: Screenshot 2023-02-08 at 12 31 18
> 
> so just go "Biosecurity and pandemic...". Not optimal, but better I think

Ah sorry I meant to put this in the PR description. I have already added a field to set a short name (e.g. Biosecurity & Pandemics) to stop them from wrapping

**Comment by Will-Howard** [comment]
> So I'd also be in favour of killing the ToC, or moving it up (which I assume is pretty complex).

Cool I'll remove it. I did try putting it at the top, which was not _that_ complex. But I just thought it looked bad and that the early centering was an ok solution

> Looks like it's now ordered by name, I think we may want to keep the order as in the excel sheet/Figma? ([link here](https://www.figma.com/file/OD4vJlLuWriHqG1WJXY4LV?node-id=447:6036#361409791) about the ordering)
👍 , I have a slight gripe about the exact ordering but I agree we should do this and we can change the order after the fact

> I'd suggest only showing 4 lines and then "Show 7 more" or "Show all"
👍

**Comment by Will-Howard** [comment]
New and improved version:

Screenshot 2023-02-08 at 15 09 00

I have now also split this out into its own file (EAAllTagsPage) because it's diverged significantly from the LW one

File: packages/lesswrong/components/tagging/AllTagsPage.tsx [hunk 5]
```diff
@@ -74,7 +85,18 @@ const AllTagsPage = ({classes}: {
   return (
     <AnalyticsContext pageContext="allTagsPage">
       <div className={classes.root}>
-        <div className={classes.topSection}>
+        {isEAForum && <div className={classes.coreTagsSection}>
```
[the remark this answers is not in the document]
Comment (Will-Howard) [reply packages/lesswrong/components/tagging/AllTagsPage.tsx]: I have now done this bc of the table of contents change

**Comment by agnesstenlund** [comment]
Looks great 🏆

**Comment by Will-Howard** [comment]
[force merging because JP is out]

**Pull request merged**